src/Mod/PartDesign/Gui/ReferenceSelection.cpp

```diff
@@ -60,11 +61,23 @@ using namespace Gui;
 
 bool ReferenceSelection::allow(App::Document* pDoc, App::DocumentObject* pObj, const char* sSubName)
 {
+    Base::Console().Log("ReferenceSelection::allow called.\n");
+    if (pObj) {
+        Base::Console().Log("  Object: %s, Type: %s, SubName: %s\n",
+                            pObj->getNameInDocument(), pObj->getTypeId().getName(), sSubName ? sSubName : "(none)");
+    }
+    else {
+        Base::Console().Log("  pObj is NULL\n");
+        return false;
+    }
+
+

```

Comment:
Reverse the condition to do an early return:
```suggestion
    Base::Console().Log("ReferenceSelection::allow called.\n");
    if (! pObj) {
        Base::Console().Log("  pObj is NULL\n");
        return false;
    }
    Base::Console().Log("  Object: %s, Type: %s, SubName: %s\n",
                            pObj->getNameInDocument(), pObj->getTypeId().getName(), sSubName ? sSubName : "(none)");

```

Are log messages really needed ?